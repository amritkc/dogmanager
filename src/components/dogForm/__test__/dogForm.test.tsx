
import { fireEvent, render, screen } from '@testing-library/react';
import DogForm from '../dogForm';


describe("App Test", () => {
    const addDogmock = jest.fn();
    test("Dog Form Renders Corectly adding dogs", () => {

        render(<DogForm addDog={addDogmock} />)

        fireEvent.change(screen.getByTestId("dogsname"), { target: { value: 'husky' } })
        fireEvent.change(screen.getByTestId("dogsage"), { target: { value: '9' } })
        fireEvent.click(screen.getByTestId("submitdogslist"));
        expect(addDogmock).toHaveBeenCalledWith("husky", 9)
    })

    test("Valididate SnapShot for DogList", () => {
        const components = render(<DogForm addDog={addDogmock} />)
        expect(components).toMatchSnapshot()
    })

})
