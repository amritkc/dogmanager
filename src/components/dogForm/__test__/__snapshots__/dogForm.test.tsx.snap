// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Valididate SnapShot for DogList 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <input
          data-testid="dogsname"
          placeholder="Dog's Name"
          type="text"
          value=""
        />
        <input
          data-testid="dogsage"
          placeholder="Dog's Age"
          type="number"
          value=""
        />
        <button
          data-testid="submitdogslist"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <input
        data-testid="dogsname"
        placeholder="Dog's Name"
        type="text"
        value=""
      />
      <input
        data-testid="dogsage"
        placeholder="Dog's Age"
        type="number"
        value=""
      />
      <button
        data-testid="submitdogslist"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
